tools/aux/tpm2_print.c

```diff
@@ -0,0 +1,336 @@
+        if (!res) {
+            goto read_error;
+        }
+    }
+
+    // print digest in hex (a TPM2B object)
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("pcrDigest: ");
+    res = tpm2_util_print_tpm2b_file(fd);
+    tpm2_tool_output("\n");
+    if (!res) {
+        goto read_error;
+    }
+
+    // success
+    return true;
+
+read_error:
+    FILE_2_SHORT_ERROR();
+error:
+    return false;
+}
+
+static bool print_TPMS_ATTEST_yaml(FILE* fd, size_t indent_count) {
+    bool res;
+
+    // print magic without converting endianness
+    UINT32 magic;
+    res = files_read_bytes(fd, (UINT8*)&magic, sizeof(UINT32));
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("magic: ");
+    tpm2_util_hexdump((const UINT8*)&magic, sizeof(UINT32), true);
+    tpm2_tool_output("\n");
+    magic = tpm2_util_ntoh_32(magic); // finally, convert endianness
+
+    // check magic
+    if (magic != TPM2_GENERATED_VALUE) {
+        LOG_ERR("%s: Bad magic", file_path);
+        goto error;
+    }
+
+    UINT16 type;
+    res = files_read_bytes(fd, (UINT8*)&type, sizeof(UINT16));
+    if (!res) {
+        goto read_error;
+    }
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("type: ");
+    tpm2_util_hexdump((const UINT8*)&type, sizeof(UINT16), true);
+    tpm2_tool_output("\n");
+    type = tpm2_util_ntoh_16(type); // finally, convert endianness
+
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("qualifiedSigner: ");
+    res = tpm2_util_print_tpm2b_file(fd);
+    tpm2_tool_output("\n");
+    if (!res) {
+        goto read_error;
+    }
+
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("extraData: ");
+    res = tpm2_util_print_tpm2b_file(fd);
+    tpm2_tool_output("\n");
+    if (!res) {
+        goto read_error;
+    }
+
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("clockInfo:\n");
+    res = print_clock_info(fd, indent_count + 1);
+    if (!res) {
+        goto error;
+    }
+
+    print_yaml_indent(indent_count);
+    tpm2_tool_output("firmwareVersion: ");
+    res = tpm2_util_hexdump_file(fd, sizeof(UINT64), true);
+    tpm2_tool_output("\n");
+    if (!res) {
+        goto read_error;
+    }
+
+    switch(type) {
+    case TPM2_ST_ATTEST_QUOTE:
+        print_yaml_indent(indent_count);
+        tpm2_tool_output("attested:\n");
+        print_yaml_indent(indent_count + 1);
+        tpm2_tool_output("quote:\n");
+        res = print_TPMS_QUOTE_INFO(fd, indent_count + 2);
+        if (!res) {
+            goto error;
+        }
+        break;
+
+    default:
+        LOG_ERR("%s: Cannot print unsupported type 0x%x", file_path, (unsigned int)type);
+        goto error;
+    }
+
+    // success
+    return true;
+
+read_error:
+    FILE_2_SHORT_ERROR();
+error:
+    return false;
+}
+
+static bool on_option(char key, char *value) {
+    switch (key) {
+    case 't':
+        if (strcmp(value, "TPMS_ATTEST") == 0) {
```

Comment:
Flipping the logic makes this simpler. If not TPMS_ATTEST, log an error and return, no need for an else clause, just assign to type.
```
case 't':
    if (strcmp(value, "TPMS_ATTEST"))  {
        LOG_ERR("msg"):
        return 1;
    }
    file_type = file_type_TPMS_ATTEST;
    break;
```

Reply:
Changed